misc/coding.md

```diff
@@ -65,31 +57,34 @@ functionalities. E.g trait helpers ends with `_t` and `_v`.
 Type and class names should start with a capital letter and have a capital letter for each new word.
 Underscores (`_`) are not allowed.
 
+Note: exceptions are allowed to follow standard library conventions in case of similar
+functionalities.
+
 The `using` syntax is preferred to `typedef` for type definitions.
 
 ### Namespace Names
 
-If possible, namespace names should be one word, lowercase.
-If multiple words are used they should be lowercase and separated with an underscores (`_`).
+In general, namespace names should be one word, lowercase.
+If multiple words are needed they should be lowercase and separated with an underscores (`_`).
 
 ### Variable Names
 
 #### Variable Names
 
 Variable names should contain only lowercase characters and underscores (`_`).
-The underscore is not allowed as first or last character.
+The underscore is used to separate words and is not allowed as first or last character.
+Two consecutive underscores (`__`) are not allowed.
 
 #### Class Data Members
 
-Class data members names follows the convention of variable names with a trailing underscore (`_`).
+Public data members names follows the convention of variable names.
+Private class data members names follows the convention of variable names with a trailing underscore (`_`).
 
 ### Function Names
 
-Function names should start with a lowercase character and have a capital letter for each new word.
-Underscores (`_`) are not allowed.
-
-The only exception to this rule are getter and setter methods for class data members.
-They follow the naming convention `set_variable` and `get_variable`, respectively (see [Class Format](#class-format)).
+Variable names should contain only lowercase characters and underscores (`_`).
+The underscore is used to separate words and is not allowed as first or last character.
+Two consecutive underscores (`__`) are not allowed.
```

Comment:
This actually has an effect on the release: the algorithm function names would be changed by this. Should we sneak this in before 0.1.0 or do it afterwards? Not the end of the world to do it after 0.1.0 since, as far as I can tell, CP2K only needs `cholesky` and `eigensolver` (not `genEigensolver`) whose names would not change by this guideline.

Reply:
My idea was to make this task part of the refactoring and not part of the release.
The update of the API will therefore happen in 0.2.0.